packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            clock.tick(25);
+            transactionStatisticsCollector.deactivate();
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            sinon.assert.calledOnce(updateSpy);
+            transactionStatisticsCollector.getRoundFinishTime().should.equal(75);
+        });
+
+    });
+
+    describe('Updating a TransactionStatisticsCollector with a TxStatus result', () => {
+
+        it('should update successful transactions', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(50);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForSuccessful().should.equal(50);
+        });
+
+        it('should update failed transactions', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusFail();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalFailedTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(50);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForFailed().should.equal(50);
```

Comment:
Maybe we should also assert that the corresponding "successful" metrics are not updated.